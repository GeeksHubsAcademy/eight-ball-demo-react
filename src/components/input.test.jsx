import React from "react";

import { shallow, mount } from "enzyme";
import Input from "./Input";

it("renders without crashing", () => {
  shallow(<Input />);
});

it("It has one input in a form", () => {
  const wrapper = shallow(<Input />);
  const form = wrapper.find("form");
  expect(form).toHaveLength(1);
  expect(form.find("input")).toHaveLength(1);
});

it("It shouldn't change", () => {
  const wrapper = shallow(<Input />);
  expect(wrapper).toMatchSnapshot();
});

it("should recieve a submit function", () => {
  const wrapper = mount(<Input submit={console.log} />);
  expect(typeof wrapper.prop("submit")).toBe("function");
});

it("should run the submit function onSubmit event", () => {
  const onSubmit = jest.fn();
  const value = "test text";
  const wrapper = mount(<Input submit={onSubmit} />);
  wrapper
    .find("input")
    .at(0)
    .simulate("change", {
      preventDefault: () => {},
      target: { value }
    });
  wrapper
    .find("form")
    .at(0)
    .simulate("submit", { preventDefault: () => {} });
  expect(onSubmit).toHaveBeenCalled();
  expect(onSubmit).toBeCalledWith(value);
});
