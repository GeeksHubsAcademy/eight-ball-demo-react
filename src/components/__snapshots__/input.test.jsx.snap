// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It shouldn't change 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="¿Qué quieres saber?"
          type="text"
          value=""
        />,
        <button
          type="submit"
        >
          Dime!
        </button>,
      ],
      "className": "Input",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "¿Qué quieres saber?",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dime!",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Dime!",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="¿Qué quieres saber?"
            type="text"
            value=""
          />,
          <button
            type="submit"
          >
            Dime!
          </button>,
        ],
        "className": "Input",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "¿Qué quieres saber?",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dime!",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Dime!",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
